pymeasure/instruments/hcp/tc038d.py

```diff
@@ -0,0 +1,136 @@
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class TC038D(Instrument):
+    """
+    Communication with the HCP TC038D oven.
+
+    This is the newer version with DC heating.
+
+    The oven expects raw bytes written, no ascii code, and sends raw bytes.
+    For the variables are two or four-byte modes available. We use the
+    four-byte mode addresses, so do we. In that case element count has to be
+    double the variables read.
+    """
+
+    functions = {'read': 0x03, 'writeMultiple': 0x10,
+                 'writeSingle': 0x06, 'echo': 0x08}
+
+    def __init__(self, resourceName, address=1, timeout=1000):
+        """Initialize the device."""
+        super().__init__(resourceName, "TC038D", timeout=timeout)
+        self.address = address
+
+    def CRC16(self, data):
+        """Calculate the CRC16 checksum for the data byte array."""
+        CRC = 0xFFFF
+        for octet in data:
+            CRC ^= octet
+            for j in range(8):
+                lsb = CRC & 0x1  # least significant bit
+                CRC = CRC >> 1
+                if lsb:
+                    CRC ^= 0xA001
+        return [CRC & 0xFF, CRC >> 8]
+
+    def readRegister(self, address, count=2):
+        """Read count variables from start address on."""
+        # Count has to be double the number of elements in 4-byte-mode.
+        data = [self.address]
+        data.append(0x03)  # function code
+        data += [address >> 8, address & 0xFF]  # 2B address
+        data += [count >> 8, count & 0xFF]  # 2B number of elements
+        data += self.CRC16(data)
+        self.adapter.connection.write_raw(bytes(data))
+        got = self.adapter.connection.read_bytes(3)
+        # Slave address, function, length
+        if got[1] == 0x03:
+            length = got[2]
+            read = self.adapter.connection.read_bytes(length + 2)
+            # data, CRC
+            return read[:-2]
+        else:  # an error occurred
+            end = self.adapter.connection.read_bytes(2)  # empty the buffer
+            if got[2] == 0x02:
+                raise ValueError("The read start address is incorrect.")
+            if got[2] == 0x03:
+                raise ValueError("The number of elements exceeds the allowed range")
+            raise ConnectionError(f"Unknown read error. Received: {got} {end}")
+
+    def writeMultiple(self, address, values, byteMode=4):
+        """Write multiple variables."""
+        data = [self.address]
+        data.append(0x10)  # function code
+        data += [address >> 8, address & 0xFF]  # 2B address
+        if isinstance(values, int):
+            data += [0x0, byteMode // 2]  # 2B number of elements
+            data.append(byteMode)  # 1B number of write data
+            for i in range(byteMode - 1, -1, -1):
+                data.append(values >> i * 8 & 0xFF)
+        elif hasattr(values, "__iter__"):
```

Comment:
`isinstance(values, collections.abc.Iterable)`?

Reply:
That was the way I found to look for an iterable. Is it preferred to use the collections?

Reply:
Either way is fine (although there are slight differences we probably won't hit) -- I guess maybe I'm personally more inclined towards isinstance.

In any case, though, are you really looking for an iterable, here? You are using `len()` on `values`, but not every iterator has a length (e.g. `count()`). Are you not looking for a *sequence*, instead?